importlib_metadata/_py39compat.py

```diff
@@ -19,30 +19,17 @@ def normalized_name(dist: Distribution) -> Optional[str]:
     except AttributeError:
         from . import Prepared  # -> delay to prevent circular imports.
 
-        return Prepared.normalize(getattr(dist, "name", None) or dist.metadata['Name'])
+        return Prepared.normalize(dist.metadata['Name'])
 
 
-def ep_matches(ep: EntryPoint, **params) -> Tuple[EntryPoint, bool]:
+def ep_matches(ep: EntryPoint, **params) -> bool:
     """
     Workaround for ``EntryPoint`` objects without the ``matches`` method.
-    For the sake of convenience, a tuple is returned containing not only the
-    boolean value corresponding to the predicate evalutation, but also a compatible
-    ``EntryPoint`` object that can be safely used at a later stage.
-
-    For example, the following sequences of expressions should be compatible:
-
-        # Sequence 1: using the compatibility layer
-        candidates = (_py39compat.ep_matches(ep, **params) for ep in entry_points)
-        [ep for ep, predicate in candidates if predicate]
-
-        # Sequence 2: using Python 3.9+
-        [ep for ep in entry_points if ep.matches(**params)]
     """
     try:
-        return ep, ep.matches(**params)
+        return ep.matches(**params)
     except AttributeError:
         from . import EntryPoint  # -> delay to prevent circular imports.
 
         # Reconstruct the EntryPoint object to make sure it is compatible.
-        _ep = EntryPoint(ep.name, ep.value, ep.group)
-        return _ep, _ep.matches(**params)
+        return EntryPoint(ep.name, ep.value, ep.group).matches(**params)
```

Comment:
There is a small risk associated with re-using the `EntryPoint` objects as returned by an old `importlib.metadata` implementation:

- If someone uses the `importlib_metadata.entry_points()`/`importlib_metadata.EntryPoints.select()` API, they expect the result to be a collection of `importlib_metadata.EntryPoint` objects.
- However, with the change what they will get is a collection of `Union[importlib_metadata.EntryPoint, importlib.metadata.EntryPoint]` which is not strictly API compatible with their expectations (and therefore may cause runtime errors).

If this risk is considered small enough, the change should be relatively safe.

Reply:
In this case, I agree the risk is low, so I'll proceed with this change.